shared/oae/api/oae.api.js

```diff
@@ -171,17 +171,120 @@ define(['oae.api.admin', 'oae.api.authentication', 'oae.api.config', 'oae.api.co
                 return;
             }
 
-            var needsToProvideDisplayName = !oae.api.util.validation().isValidDisplayName(oae.data.me.displayName);
-            var needsToProvideEmail = !oae.data.me.email;
+            // Perform any invitation accepting if instructed
+            acceptInvitation(location, function(resources) {
+                // Perform any email verification if instructed before we try and determine if the
+                // user's profile info is valid
+                verifyEmail(location, function() {
+                    var needsToProvideDisplayName = !oae.api.util.validation().isValidDisplayName(oae.data.me.displayName);
+                    var needsToProvideEmail = !oae.data.me.email;
+
+                    // Show the edit profile widget if a valid name or email address need to be provided
+                    if (needsToProvideDisplayName || needsToProvideEmail) {
+                        $(document).trigger('oae.trigger.editprofile');
+
+                    // Show the Terms and Conditions widget if the user needs to accept the Terms and Conditions
+                    } else if (oae.data.me.needsToAcceptTC) {
+                        oae.api.widget.insertWidget('termsandconditions', null, null, true);
+                    }
+                });
+            });
+        };
+
+        /**
+         * Accept the invitation if the query string of the specified location indicates there is an
+         * invitation to be accepted
+         *
+         * @param  {String}         location            The URL whose location to check for invitation info
+         * @param  {Function}       callback            Invoked when the accept invitation request is complete
```

Comment:
It's not a standard callback function, it is in charge of its own error handling (notification), and it returns a valid response as the first parameter.